olive/data/component/load_dataset.py

```diff
@@ -27,6 +27,11 @@ def huggingface_dataset(data_dir, data_name=None, subset=None, split="validation
     """Create a dataset from huggingface datasets."""
     from datasets.utils.logging import disable_progress_bar, set_verbosity_error
 
+    if kwargs.get("huggingface_token") is not None:
+        from huggingface_hub import login
+
+        login(token=kwargs["huggingface_token"])
```

Comment:
we will pass the token as text in aml job? are there any security concerns?

Reply:
For the choice: save the token under `~/.cache/huggingface/token`, somehow it can be done by user out of Olive.

After set it in cache file, `get_token_permission` can return the right results even without token passed.
![image](https://github.com/microsoft/Olive/assets/13343117/4e38d885-f716-453e-95b5-290e31ca772e)

Reply:
@trajepl thanks for the heads up. Instead, I prefer directly getting token from local huggingface cache. I can make this change.